### ISSUE TITLE
hlint tests break with hlint-2.0

### ISSUE THREAD

**Issue by RyanGlScott** [ekmett/lens#723]
The `hlint` test suite has a quite a few failures when ran against `hlint-2.0`. [From a recent broken Travis run:](https://travis-ci.org/ekmett/lens/jobs/219499516#L1137)

```
Test suite hlint: RUNNING...
Warning: Can't find file "HsBaseConfig.h" in directories
	src/Control/Lens
	.
  Asked for by: src/Control/Lens/Wrapped.hs  at line 85 col 1
src/Control/Lens/Lens.hs:1478:1: Warning: Redundant lambda
Found:
  fusing t = \ f -> lowerYoneda . t (liftYoneda . f)
Why not:
  fusing t f = lowerYoneda . t (liftYoneda . f)

src/Control/Lens/Reified.hs:15:1: Warning: Use module export list
Found:
  module Control.Lens.Reified where
Why not:
  module Control.Lens.Reified (module Control.Lens.Reified) where
Note: An explicit list is usually better

src/Control/Lens/Reified.hs:140:26: Suggestion: Use const
Found:
  \ _ -> a
Why not:
  const a

src/Control/Lens/Reified.hs:371:29: Suggestion: Use const
Found:
  \ _ -> [a]
Why not:
  const [a]

src/Control/Lens/TH.hs:682:3: Suggestion: Reduce duplication
Found:
  method <- computeMethod fieldPart
  let cls = "Has" ++ fieldPart
  return (MethodName (mkName cls) (mkName method))
Why not:
  Combine with src/Control/Lens/TH.hs:729:3

src/Control/Lens/Tuple.hs:1068:1: Warning: Redundant lambda
Found:
  _1' = \ f !x -> _1 f x
Why not:
  _1' f !x = _1 f x

src/Control/Lens/Tuple.hs:1073:1: Warning: Redundant lambda
Found:
  _2' = \ f !x -> _2 f x
Why not:
  _2' f !x = _2 f x

src/Control/Lens/Tuple.hs:1078:1: Warning: Redundant lambda
Found:
  _3' = \ f !x -> _3 f x
Why not:
  _3' f !x = _3 f x

src/Control/Lens/Tuple.hs:1083:1: Warning: Redundant lambda
Found:
  _4' = \ f !x -> _4 f x
Why not:
  _4' f !x = _4 f x

src/Control/Lens/Tuple.hs:1088:1: Warning: Redundant lambda
Found:
  _5' = \ f !x -> _5 f x
Why not:
  _5' f !x = _5 f x

src/Control/Lens/Tuple.hs:1093:1: Warning: Redundant lambda
Found:
  _6' = \ f !x -> _6 f x
Why not:
  _6' f !x = _6 f x

src/Control/Lens/Tuple.hs:1098:1: Warning: Redundant lambda
Found:
  _7' = \ f !x -> _7 f x
Why not:
  _7' f !x = _7 f x

src/Control/Lens/Tuple.hs:1103:1: Warning: Redundant lambda
Found:
  _8' = \ f !x -> _8 f x
Why not:
  _8' f !x = _8 f x

src/Control/Lens/Tuple.hs:1108:1: Warning: Redundant lambda
Found:
  _9' = \ f !x -> _9 f x
Why not:
  _9' f !x = _9 f x

src/Control/Lens/Tuple.hs:1113:1: Warning: Redundant lambda
Found:
  _10' = \ f !x -> _10 f x
Why not:
  _10' f !x = _10 f x

src/Control/Lens/Tuple.hs:1118:1: Warning: Redundant lambda
Found:
  _11' = \ f !x -> _11 f x
Why not:
  _11' f !x = _11 f x

src/Control/Lens/Tuple.hs:1123:1: Warning: Redundant lambda
Found:
  _12' = \ f !x -> _12 f x
Why not:
  _12' f !x = _12 f x

src/Control/Lens/Tuple.hs:1128:1: Warning: Redundant lambda
Found:
  _13' = \ f !x -> _13 f x
Why not:
  _13' f !x = _13 f x

src/Control/Lens/Tuple.hs:1133:1: Warning: Redundant lambda
Found:
  _14' = \ f !x -> _14 f x
Why not:
  _14' f !x = _14 f x

src/Control/Lens/Tuple.hs:1138:1: Warning: Redundant lambda
Found:
  _15' = \ f !x -> _15 f x
Why not:
  _15' f !x = _15 f x

src/Control/Lens/Tuple.hs:1143:1: Warning: Redundant lambda
Found:
  _16' = \ f !x -> _16 f x
Why not:
  _16' f !x = _16 f x

src/Control/Lens/Tuple.hs:1148:1: Warning: Redundant lambda
Found:
  _17' = \ f !x -> _17 f x
Why not:
  _17' f !x = _17 f x

src/Control/Lens/Tuple.hs:1153:1: Warning: Redundant lambda
Found:
  _18' = \ f !x -> _18 f x
Why not:
  _18' f !x = _18 f x

src/Control/Lens/Tuple.hs:1158:1: Warning: Redundant lambda
Found:
  _19' = \ f !x -> _19 f x
Why not:
  _19' f !x = _19 f x

src/Control/Lens/Internal/Indexed.hs:205:20: Suggestion: Use const
Found:
  \ _ -> f
Why not:
  const f

src/Control/Lens/Internal/PrismTH.hs:293:3: Suggestion: Reduce duplication
Found:
  conName = view nconName con
  fields = length (view nconTypes con)
  reviewer = makeReviewer conName fields
Why not:
  Combine with src/Control/Lens/Internal/PrismTH.hs:306:3

src/Control/Lens/Internal/TH.hs:35:1: Warning: Use module export list
Found:
  module Control.Lens.Internal.TH where
Why not:
  module Control.Lens.Internal.TH (module Control.Lens.Internal.TH)
         where
Note: An explicit list is usually better

src/Control/Lens/Internal/Zoom.hs:328:45: Suggestion: Use >=>
Found:
  \ st ->
    m st >>=
      \ (s, t, w) ->
        n t >>= \ (s', u, w') -> return (mappend s s', u, mappend w w')
Why not:
  (m Control.Monad.>=>
     (\ (s, t, w) ->
        n t >>= \ (s', u, w') -> return (mappend s s', u, mappend w w')))

src/System/IO/Error/Lens.hs:14:1: Warning: Use module export list
Found:
  module System.IO.Error.Lens where
Why not:
  module System.IO.Error.Lens (module System.IO.Error.Lens) where
Note: An explicit list is usually better

29 hints
Test suite hlint: FAIL
Test suite logged to: dist/test/lens-4.15.1-hlint.log
```

Curiously, I believe some of these suggestions (e.g., Redundant lambda) are things which should be ignored, according to `HLint.hs`. Perhaps `hlint-2.0` changed the way it configures custom settings?

**Comment by ndmitchell** [comment]
A bug in hlint-2.0, fixed in hlint-2.0.1. I've also raised #724 which reenables the newer version.

**Comment by RyanGlScott** [comment]
Thanks for the quick fix, @ndmitchell!
